src/authentication/abax-auth.ts

```diff
@@ -51,16 +51,27 @@ export interface AbaxAuthConfig {
    * @default https://identity.abax.cloud
    */
   baseUrl?: string | undefined;
+  /**
+   * The minimum lifetime of the token in seconds. If the token expires in less
+   * than this time, it will be refreshed.
+   */
+  minimumTokenLifetime?: number | undefined;
 }
 
 export class AbaxAuth {
   private baseUrl = 'https://identity.abax.cloud';
   private credentials?: AbaxCredentials;
+  private minimumTokenLifetime: number;
 
   constructor(private readonly config: AbaxAuthConfig) {
```

Comment:
```suggestion
  constructor(private readonly {baseUrl, minimumTokenLifetime = 60 }: AbaxAuthConfig) {
```

Could something like this work? Feels awkward to have an if statement when we are just applying a default value. 

We should also prevent the user from setting a negative value and breaking things.

Reply:
Updated to use `??` operator, as the `{variables}` in constructor was not a valid with error:
```
A parameter property may not be declared using a binding pattern.ts(1187)
```